frontend/packages/pipelines-plugin/src/components/shared/workspaces/VolumeClaimTemplateLink.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { PersistentVolumeClaimKind } from '@console/internal/module/k8s';
+import { PersistentVolumeClaimModel } from '@console/internal/models';
+import { ResourceLink } from '@console/internal/components/utils';
+import { getMatchedPVCs } from '../../../utils/pipeline-utils';
+
+export interface VolumeClaimTemplateLinkProps {
+  labels: { [key: string]: string };
+  workspaceName: string;
+  namespace: string;
+  ownerReference: string;
+}
+
+const VolumeClaimTemplateLink: React.FC<VolumeClaimTemplateLinkProps> = ({
+  labels,
+  workspaceName,
+  namespace,
+  ownerReference,
+}) => {
+  const [pvcResources, loaded, loadError] = useK8sWatchResource<PersistentVolumeClaimKind[]>({
+    kind: PersistentVolumeClaimModel.kind,
+    isList: true,
+    selector: { matchLabels: labels },
+  });
```

Comment:
And you need to constrain this on `namespace`

Reply:
added namespace